src/blockchain.js

```diff
@@ -1,3 +1,131 @@
+
+function reduce(arr, memo, iteratee, cb) {
```

Comment:
You know that ES6 has it's own Array.prototype.reduce` that does pretty much the same thing, just without the callback.
Eg: https://github.com/embark-framework/embark/blob/develop/lib/utils/utils.js#L422